ocs_ci/ocs/bucket_utils.py

```
@@ -3651,6 +3651,76 @@ def get_noobaa_bucket_replication_metrics_in_prometheus(
+            if check_not_optimal and mode not in ("OPTIMAL", "UNKNOWN"):
+                logger.info(f"Bucket {bucket_name} entered non-OPTIMAL mode: {mode}")
+                return mode
+            elif not check_not_optimal and mode == expected_mode:
+                logger.info(f"Bucket {bucket_name} entered expected mode: {mode}")
+                return mode
+    except TimeoutExpiredError:
+        if check_not_optimal:
+            assert False, (
+                f"Bucket {bucket_name} is still OPTIMAL after "
+                f"{timeout}s (last mode: {last_mode})"
+            )
```

Ruff (0.15.17)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)